src/handler/mod.rs

```diff
@@ -760,14 +835,18 @@ impl Handler {
         );
 
         if let Some(challenge) = self.active_challenges.remove(&node_address) {
+            let local_key = self.key.clone();
+            let local_id = self.node_id;
             match Session::establish_from_challenge(
-                self.key.clone(),
-                &self.node_id,
+                local_key,
+                &local_id,
                 &node_address.node_id,
```

Comment:
```rust
        if let Some(challenge) = self.active_challenges.remove(&node_address) {
+           let local_key = self.key.clone();
+           let local_id = self.node_id;
            match Session::establish_from_challenge(
-               self.key.clone(),
-               &self.node_id,
+               local_key,
+               &local_id,
                &node_address.node_id,
```
Does local_key and local_id have to be pulled out for a reason?

Reply:
No actually not, good catch! I did that because I was going to pass a mutable reference to `Handler<P>` instead of just the mutable reference to its filed `sessions`. Download rust-analyzer and let it tell you why that would have been a problem without assigning the `key` and `node_id` to variables like in the code above.